src/internal/m365/onedrive/restore_test.go

```diff
@@ -321,28 +322,73 @@ func (suite *RestoreUnitSuite) TestAugmentRestorePaths_DifferentRestorePath() {
 	}
 }
 
-func (suite *RestoreUnitSuite) TestRestoreItem_errItemAlreadyExists() {
+func (suite *RestoreUnitSuite) TestRestoreItem_collisionHandling() {
 	table := []struct {
 		name          string
+		collisionKeys map[string]string
 		onCollision   control.CollisionPolicy
-		expectErr     func(*testing.T, error)
 		expectSkipped assert.BoolAssertionFunc
+		expectMock    func(*testing.T, *mock.RestoreHandler)
 	}{
 		{
-			name:        "skip",
-			onCollision: control.Skip,
-			expectErr: func(t *testing.T, err error) {
-				require.NoError(t, err, clues.ToCore(err))
+			name:          "no collision, copy",
+			collisionKeys: map[string]string{},
+			onCollision:   control.Copy,
+			expectSkipped: assert.False,
+			expectMock: func(t *testing.T, rh *mock.RestoreHandler) {
+				assert.True(t, rh.CalledPostItem, "new item posted")
+				assert.False(t, rh.CalledDeleteItem, "new item deleted")
+			},
+		},
+		{
+			name:          "no collision, replace",
+			collisionKeys: map[string]string{},
+			onCollision:   control.Replace,
+			expectSkipped: assert.False,
+			expectMock: func(t *testing.T, rh *mock.RestoreHandler) {
+				assert.True(t, rh.CalledPostItem, "new item posted")
+				assert.False(t, rh.CalledDeleteItem, "new item deleted")
 			},
-			expectSkipped: assert.True,
 		},
 		{
-			name:        "replace",
-			onCollision: control.Replace,
-			expectErr: func(t *testing.T, err error) {
-				require.ErrorIs(t, err, graph.ErrItemAlreadyExistsConflict, clues.ToCore(err))
+			name:          "no collision, skip",
+			collisionKeys: map[string]string{},
+			onCollision:   control.Skip,
+			expectSkipped: assert.False,
+			expectMock: func(t *testing.T, rh *mock.RestoreHandler) {
+				assert.True(t, rh.CalledPostItem, "new item posted")
+				assert.False(t, rh.CalledDeleteItem, "new item deleted")
+			},
+		},
+		{
+			name:          "collision, copy",
+			collisionKeys: map[string]string{mock.DriveItemFileName: "smarf"},
+			onCollision:   control.Copy,
+			expectSkipped: assert.False,
+			expectMock: func(t *testing.T, rh *mock.RestoreHandler) {
+				assert.True(t, rh.CalledPostItem, "new item posted")
+				assert.False(t, rh.CalledDeleteItem, "new item deleted")
 			},
+		},
+		{
+			name:          "collision, replace",
+			collisionKeys: map[string]string{mock.DriveItemFileName: "smarf"},
+			onCollision:   control.Replace,
 			expectSkipped: assert.False,
+			expectMock: func(t *testing.T, rh *mock.RestoreHandler) {
+				assert.True(t, rh.CalledPostItem, "new item posted")
+				assert.True(t, rh.CalledDeleteItem, "new item deleted")
```

Comment:
we should probably be checking the ID of the thing we're deleting

Reply:
can do